compiler/qsc_hir/src/hir.rs

```diff
@@ -215,10 +215,25 @@ impl Display for Res {
 pub struct Package {
     /// The items in the package.
     pub items: IndexMap<LocalItemId, Item>,
+    /// The top-level statements in the package.
+    pub stmts: Vec<Stmt>,
     /// The entry expression for an executable package.
     pub entry: Option<Expr>,
 }
 
+impl Package {
+    /// Extends the `Package` with the contents of another `Package`.
+    /// `other` should not contain any `LocalItemId`s
+    /// that conflict with the current `Package`.
```

Comment:
Same thing here, it's worth using the `Validator` (the hir variant) after the packages have been merged.